src/converter/one/convert/JfrToOtlp.java

```diff
@@ -0,0 +1,184 @@
+
+        out.write(otlpProto.buffer(), 0, otlpProto.size());
+    }
+
+    private void writeProfileDictionary() {
+        int profilesDictionaryMark = otlpProto.startField(Otlp.ProfilesData.DICTIONARY);
+
+        int mappingMark = otlpProto.startField(Otlp.ProfilesDictionary.MAPPING_TABLE);
+        otlpProto.commitField(mappingMark);
+
+        // Write function table
+        for (String functionName : functionPool.keys()) {
+            int functionMark = otlpProto.startField(Otlp.ProfilesDictionary.FUNCTION_TABLE);
+            otlpProto.field(Otlp.Function.NAME_STRINDEX, stringPool.index(functionName));
+            otlpProto.commitField(functionMark);
+        }
+
+        // Write location table
+        for (long functionIdx = 0; functionIdx < functionPool.size(); ++functionIdx) {
+            int locationMark = otlpProto.startField(Otlp.ProfilesDictionary.LOCATION_TABLE);
+            otlpProto.field(Otlp.Location.MAPPING_INDEX, 0);
+            int lineMark = otlpProto.startField(Otlp.Location.LINE);
+            otlpProto.field(Otlp.Line.FUNCTION_INDEX, functionIdx);
+            otlpProto.commitField(lineMark);
+            otlpProto.commitField(locationMark);
+        }
+
+        // Write string table
+        for (String s : stringPool.keys()) {
+            otlpProto.field(Otlp.ProfilesDictionary.STRING_TABLE, s);
+        }
+
+        otlpProto.commitField(profilesDictionaryMark);
+    }
+
+    public static void convert(String input, String output, Arguments args) throws IOException {
+        JfrToOtlp converter;
+        try (JfrReader jfr = new JfrReader(input)) {
+            converter = new JfrToOtlp(jfr, args);
+            converter.convert();
+        }
+        try (FileOutputStream out = new FileOutputStream(output)) {
+            converter.dump(out);
+        }
+    }
+
+    private final class OtlpEventVisitor extends NormalizedEventVisitor {
+        private final List<SampleInfo> sampleInfos;
+
+        public OtlpEventVisitor(List<SampleInfo> sampleInfos) {
+            this.sampleInfos = sampleInfos;
+        }
+
+        @Override
+        public void visitImpl(Event event, long samples, long value) {
+            StackTrace stackTrace = jfr.stackTraces.get(event.stackTraceId);
+            if (stackTrace == null) {
+                return;
+            }
+
+            Arguments args = JfrToOtlp.this.args;
+            long[] methods = stackTrace.methods;
+            byte[] types = stackTrace.types;
+            int[] locations = stackTrace.locations;
+
+            for (int i = methods.length; --i >= 0; ) {
+                String methodName = getMethodName(methods[i], types[i]);
+                int location;
+                if (args.lines && (location = locations[i] >>> 16) != 0) {
+                    methodName += ":" + location;
+                } else if (args.bci && (location = locations[i] & 0xffff) != 0) {
+                    methodName += "@" + location;
+                }
```

Comment:
It's unnatural to have line numbers included in method names when the format provides a dedicated field for line numbers.